### PR TITLE
UI: Show a message in the empty source list

### PR DESCRIPTION
**Pull request by DDRBoxman**
Accidentally hit -d instead of -f 

see #1766

### CODE REVIEW

**Comment by DDRBoxman** [comment]
Screen Shot 2019-03-21 at 8 48 58 PM

**Comment by SuslikV** [comment]
I think that the @dodgepong suggestion (about the arrow pointing to the button), if it shown when "no sources" warning rises, is better solution. It is possible to implement it through the splash screen of Qt. It may has short message, appearance can be animated (fade, position etc.) with relatively small amount of code changes.

Also, the _Sources_ pane can be easily closed by user right now. _Preview_ window can be disabled and there is no menu entry (main menu can't be hidden right now in Windows version of the application) where you can surely add the source to the current scene.

There is one more thing. If the list is empty and shown - every human will like to fill it up. It's like magnets on refrigerator. And if picture is ugly - you'll want to replace it with something more comfortable for you. These 4 miniatures below the message will be clicked first, I think. Because they are bigger than even tool buttons that should be clicked instead.

File: UI/window-basic-main.cpp [hunk 1]
```diff
@@ -3181,8 +3192,9 @@ void OBSBasic::RemoveSelectedSceneItem()
 	OBSSceneItem item = GetCurrentSceneItem();
 	if (item) {
 		obs_source_t *source = obs_sceneitem_get_source(item);
-		if (QueryRemoveSource(source))
+		if (QueryRemoveSource(source)) {
```
Comment (WizardCM) [inline UI/window-basic-main.cpp]: This change seems unnecessary and unneeded (IDE?).

**Comment by Fenrirthviti** [comment]
Some comments:

I agree that this should also have some kind of graphic that is displayed in the preview, as well as the Sources menu. In that regard, some kind of "How to get started" splash screen with the 4 quickstart steps summarized might be useful to put in the preview until a source is added.

I don't think we need an arrow pointing to the specific + icon, the current text feels fine.

While this is likely a separate PR/discussion, it is very related to this. If we are going to add graphics/icons to the add a source message, we should implement icons in the add source menu as well. One proposal was:

add-souce 

Which I rather like. It eliminates the need for a context menu, and makes adding sources in general significantly more user friendly.

**Comment by jp9000** [comment]
I changed this to actually draw the message and image on the widget instead of use a separate widget on top.  The separate widget was causing some layout issues once items were added/removed.  For example, if you started from 0 and then added a source, the bottom section of the window would suddenly change in size.  I figure just drawing on the widget itself would be sufficient.

**Comment by SuslikV** [comment]
I want to say few words about the picture posted above, @Fenrirthviti , I don't understand where to click to, for example, when I'm looking at this window.

**Pull request merged**